src/Extensions/AzureBlobPayloads/Converters/BlobPayloadStore.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Globalization;
+using System.IO.Compression;
+using System.Text;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace Microsoft.DurableTask.Converters;
+
+/// <summary>
+/// Azure Blob Storage implementation of <see cref="IPayloadStore"/>.
+/// Stores payloads as blobs and returns opaque tokens in the form "blob:v1:&lt;container&gt;:&lt;blobName&gt;".
+/// </summary>
+public sealed class BlobPayloadStore : IPayloadStore
+{
+    readonly BlobContainerClient containerClient;
+    readonly LargePayloadStorageOptions options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BlobPayloadStore"/> class.
+    /// </summary>
+    /// <param name="options">The options for the blob payload store.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="options"/> is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when <paramref name="options.ConnectionString"/> is null or empty.</exception>
+    public BlobPayloadStore(LargePayloadStorageOptions options)
+    {
+        this.options = options ?? throw new ArgumentNullException(nameof(options));
+
+        Check.NotNullOrEmpty(options.ConnectionString, nameof(options.ConnectionString));
+        Check.NotNullOrEmpty(options.ContainerName, nameof(options.ContainerName));
+
+        BlobServiceClient serviceClient = new(options.ConnectionString);
+        this.containerClient = serviceClient.GetBlobContainerClient(options.ContainerName);
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> UploadAsync(ReadOnlyMemory<byte> payloadBytes, CancellationToken cancellationToken)
+    {
+        // Ensure container exists
+        await this.containerClient.CreateIfNotExistsAsync(PublicAccessType.None, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
nit: in this repo we try to keep line length to 120 or less. One simple thing you can do is remove the `cancellationToken: ` argument prefix since it's redundant with the variable name already. You can also considering removing `.ConfigureAwait(false)` since I usually try to avoid using this.

Otherwise, if you can't keep a line to 120 or less, you should wrap the parameter list to be a multi-line list.

Reply:
updated